ggml/src/ggml-sycl.cpp

```diff
@@ -5167,38 +5169,182 @@ static bool ggml_backend_sycl_supports_op(ggml_backend_t backend, const ggml_ten
             return false;
     }
 
-    UNUSED(backend);
+    UNUSED(dev);
 }
 
-static bool ggml_backend_sycl_offload_op(ggml_backend_t backend, const ggml_tensor * op) {
+static bool ggml_backend_sycl_device_offload_op(ggml_backend_dev_t dev, const ggml_tensor * op) {
     const int min_batch_size = 32;
     return op->ne[1] >= min_batch_size && op->op != GGML_OP_GET_ROWS && op->op != GGML_OP_MUL_MAT_ID;
-    GGML_UNUSED(backend);
+    GGML_UNUSED(dev);
+}
+
+static void ggml_backend_sycl_event_record(ggml_backend_t backend, ggml_backend_event_t event)
+try
+{
+    if (event == nullptr || event->context == nullptr) {
```

Comment:
Dumb question, is this silent exit when no events provided intended? I wonder if this could hide other issues if this function is not called when intended.

Reply:
Yes it's tricky, I went with the simple interpretation that a null event "should be ignored" consistently across all event APIs, nothing to sync/record etc.. This is currently used in model loading only and cannot be captured in other tests, worst case scenario would likely be a performance drop in model loading due to messed scheduling/synchronizations. Not sure what's the ideal thing to do here : 
- Remove the silent exit altogether.
- Add a `GGML_SYCL_DEBUG` at that level. 

What do you think @joeatodd ?

Reply:
> This is currently used in model loading only

Events will also be used for pipeline parallelism when using multiple GPUs, you should verify that it works correctly in that case.

Reply:
Thanks for the reminder @slaren noted!